src/server/game/Spells/SpellInfo.cpp

```diff
@@ -1775,13 +1775,11 @@ SpellCastResult SpellInfo::CheckTarget(Unit const* caster, WorldObject const* ta
             return SPELL_FAILED_TARGET_AFFECTING_COMBAT;
 
         // only spells with SPELL_ATTR3_ONLY_ON_GHOSTS can target ghosts
-        if (((IsRequiringDeadTarget() != 0) != unitTarget->HasAuraType(SPELL_AURA_GHOST)) && !(IsDeathPersistent() && IsAllowingDeadTarget()))
-        {
-            if (AttributesEx3 & SPELL_ATTR3_ONLY_ON_GHOSTS)
```

Comment:
What about spells with this attribute - do they still work correctly?

Reply:
Responded on Discord

Reply:
IsAllowingDeadTarget() checks that attribute. It was a double check before